REGRESSION(262844@main): css3/filters/reference-filter-change-repaint.html is randomly failing

cdumez opened the pull request:
#### 2e76ca3e586aac463212f6e0bcb0721fe41596d8
<pre>
REGRESSION(262844@main): css3/filters/reference-filter-change-repaint.html is randomly failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=255383">https://bugs.webkit.org/show_bug.cgi?id=255383</a>

Reviewed by Said Abou-Hallawa.

262844@main slightly changed the ordering of attribute processing in SVGFETurbulenceElement
and it was caused this repaint test to become flaky. We need to make sure that
SVGFETurbulenceElement::attributeChanged() calls this first:
```
m_seed-&gt;setBaseValInternal(newValue.toFloat());
```
when the seed attribute changes, before calling the base class&apos;s attributeChanged.

This patch restores the ordering to address the issue.

* Source/WebCore/svg/SVGFETurbulenceElement.cpp:
(WebCore::SVGFETurbulenceElement::attributeChanged):

Canonical link: <a href="https://commits.webkit.org/262985@main">https://commits.webkit.org/262985@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0da5a72692d7a8c2922af60b313e587536fa8efa

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3017 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3082 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3186 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4428 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3414 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/2986 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3162 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3123 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2637 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3048 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3403 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2707 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4223 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/900 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2686 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/2535 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2674 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2737 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3963 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3085 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/2475 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2713 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2688 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2703 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/374 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2904 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

shallawa reviewed (approved):
The change seems fine. But how can we make sure 262844@main did not change the order of processing the attributes of something else? Or is this change going to affect layout tests only?

cdumez committed:
REGRESSION(262844@main): css3/filters/reference-filter-change-repaint.html is randomly failing
https://bugs.webkit.org/show_bug.cgi?id=255383

Reviewed by Said Abou-Hallawa.

262844@main slightly changed the ordering of attribute processing in SVGFETurbulenceElement
and it was caused this repaint test to become flaky. We need to make sure that
SVGFETurbulenceElement::attributeChanged() calls this first:
```
m_seed->setBaseValInternal(newValue.toFloat());
```
when the seed attribute changes, before calling the base class's attributeChanged.

This patch restores the ordering to address the issue.

* Source/WebCore/svg/SVGFETurbulenceElement.cpp:
(WebCore::SVGFETurbulenceElement::attributeChanged):

Canonical link: https://commits.webkit.org/262985@main
